### PR TITLE
upgrade to typespec 0.44.0

### PR DESCRIPTION
**Pull request by chunyu3**
# Description
Fix https://github.com/Azure/autorest.csharp/issues/3403

- upgrade typespec to 0.44.0

# Checklist

To ensure a quick review and merge, please ensure:
- [ ] The PR has a understandable title and description explaining the _why_ and _what_.
- [ ] The PR is opened in draft if not ready for review yet.
   - If opened in draft, please allocate sufficient time (24 hours) after moving out of draft for review
- [ ] The branch is recent enough to not have merge conflicts upon creation.

# Ready to Land?
- [ ] Build is completely green
   - Submissions with test failures require tracking issue and approval of a CODEOWNER
- [ ] At least one +1 review by a CODEOWNER
- [ ] All -1 reviews are confirmed resolved by the reviewer 
   - Override/Marking reviews stale must be discussed with CODEOWNERS first

### CODE REVIEW

**Review by pshao25 — APPROVED** [review]

File: src/CADL.Extension/Emitter.Csharp/package.json [hunk 1]
```diff
@@ -50,17 +50,17 @@
...
-        "@azure-tools/typespec-client-generator-core": "0.30.0-dev.13",
-        "@typespec/compiler": "0.43.0",
-        "@typespec/eslint-config-typespec": "0.6.0",
-        "@typespec/eslint-plugin": "0.43.0",
-        "@typespec/http": "0.43.1",
-        "@typespec/library-linter": "0.43.0",
-        "@typespec/openapi": "0.43.0",
-        "@typespec/rest": "0.43.0",
-        "@typespec/versioning": "0.43.0",
+        "@azure-tools/typespec-autorest": "0.30.0",
```
Comment (lirenhe) [inline src/CADL.Extension/Emitter.Csharp/package.json:53]: Can we move to 'use peerDependencies with a range'?  Is there any blocker in dotnet?

Comment (lirenhe) [reply src/CADL.Extension/Emitter.Csharp/package.json:53]: You may refer https://github.com/Azure/autorest.java/blob/main/typespec-extension/package.json#L47-L53

Comment (chunyu3) [reply src/CADL.Extension/Emitter.Csharp/package.json:53]: We have not make sure peerDependencies works for our project.  We will not implement `use peerDependencies with a range` in this PR.

Comment (lirenhe) [reply src/CADL.Extension/Emitter.Csharp/package.json:53]: I see. I won't block the merge of this PR, but please create an issue to track the work of enabling PeerDependency.

**Comment by lirenhe** [comment]
Do you have the SDK regen PR?

**Review by lirenhe — APPROVED** [review]

**Pull request merged**